src/gt4py/analysis/passes.py

```diff
@@ -561,27 +564,31 @@ def write_after_read_fields_in(self, target: "MultiStageMergingWrapper") -> Set[
         current_writes = set(self.outputs)
         return previous_reads.intersection(current_writes)
 
-    def read_after_write_extents_in(self, target: "MultiStageMergingWrapper") -> Set[Extent]:
-        return {self.inputs[name] for name in self.read_after_write_fields_in(target)}
+    def compute_access_offset_for(self, fields: Iterable[str]) -> Set[Extent]:
+        # TODO This may still be incorrect when there are horizontal ifs
+        compute_extents = {field: Extent.zeros() for field in fields}
+        for ij_block in self.ij_blocks:
+            for int_block in ij_block.interval_blocks:
+                for stmt_info in int_block.stmts:
+                    for field in {f for f in fields if f in stmt_info.inputs}:
+                        compute_extents[field] |= ij_block.compute_extent + stmt_info.inputs[field]
```

Comment:
This can and should be refactored to use a generator function, but the core problem is that the infos are computing the input and output extents in not quite the right way. They are `OR`-ing each usage, which is I think more careful but less accurate than what I propose doing, which is shifting it by the compute extents then `OR`-ing only within the IJ block. Could you verify that this is indeed what we want?

If it is what we should be doing, then I propose we move this logic into the correct places in `BlockMaker.visit_ComputationBlock` and `MultiStageMergingWrapper.merge_with`.